hazelcast/src/main/java/com/hazelcast/query/impl/OrderedIndexStore.java

```diff
@@ -115,11 +116,19 @@ public boolean canEvaluate(Class<? extends Predicate> predicateClass) {
     }
 
     @Override
-    public Iterator<QueryableEntry> getSqlRecordIterator() {
-        Iterator<QueryableEntry> iterator = new IndexEntryFlatteningIterator(recordMap.values().iterator());
-        Iterator<QueryableEntry> nullIterator = recordsWithNullValue.values().iterator();
+    public Iterator<QueryableEntry> getSqlRecordIterator(boolean descending) {
```

Comment:
Here we have `ASC NULLS FIRST` and `DESC NULLS LAST`. Could you please confirm that it works the same way for :
1. Composite indexes (e.g. `[x, NULL]` comes before '[x, y]' for `ASC`, and vice versa for `DESC`)
2. Merge sorter
3. `IN` filter